core/src/test/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -312,6 +312,25 @@ public static double randomDouble() {
         return random().nextDouble();
     }
 
+    /**
+     * Returns a double value in the interval [start, end) if lowerInclusive is set to true,
+     * (start, end) otherwise.
+     *
+     * @param start lower bound of interval to draw uniformly distributed random numbers from
+     * @param end upper bound
+     * @param lowerInclusive whether or not to include lower end of the interval
+     * */
+    public static double randomDoubleBetween(double start, double end, boolean lowerInclusive) {
```

Comment:
I think the idea here is that this is general enough for ESTestCase, there is already a `randomIntBetween()` helper and this might be useful for other tests in the future as well. I'd like to keep it there to be honest.